repo/fsrepo/fsrepo.go

```diff
@@ -240,14 +416,39 @@ func (r *FSRepo) SetConfigKey(key string, value interface{}) error {
 	if r.state != opened {
 		return debugerror.Errorf("repo is %s", r.state)
 	}
-	return r.configComponent.SetConfigKey(key, value)
+
+	filename, err := config.Filename(r.path)
+	if err != nil {
+		return err
+	}
+	switch v := value.(type) {
+	case string:
+		if i, err := strconv.Atoi(v); err == nil {
+			value = i
+		}
+	}
+	var mapconf map[string]interface{}
+	if err := serialize.ReadConfigFile(filename, &mapconf); err != nil {
+		return err
+	}
+	if err := common.MapSetKV(mapconf, key, value); err != nil {
+		return err
+	}
+	conf, err := config.FromMap(mapconf)
+	if err != nil {
+		return err
+	}
+	if err := serialize.WriteConfigFile(filename, mapconf); err != nil {
+		return err
+	}
```

Comment:
btw, if you were wondering why the weird way to write out the config-- the complicating piece here is the user (or other programs) may have added keys to the config that we -- in our types -- dont know about and must not nuke. (our config works like git's).

actually, may be worth adding a comment noting that